exporter/internal/otlptext/profiles.go

```diff
@@ -0,0 +1,43 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otlptext // import "go.opentelemetry.io/collector/exporter/internal/otlptext"
+
+import "go.opentelemetry.io/collector/pdata/pprofile"
+
+// NewTextProfilesMarshaler returns a pprofile.Marshaler to encode to OTLP text bytes.
+func NewTextProfilesMarshaler() pprofile.Marshaler {
+	return textProfilesMarshaler{}
+}
+
+type textProfilesMarshaler struct{}
+
+// MarshalProfiles pprofile.Profiles to OTLP text.
+func (textProfilesMarshaler) MarshalProfiles(pd pprofile.Profiles) ([]byte, error) {
+	buf := dataBuffer{}
+	rps := pd.ResourceProfiles()
+	for i := 0; i < rps.Len(); i++ {
+		buf.logEntry("ResourceProfiles #%d", i)
+		rp := rps.At(i)
+		buf.logEntry("Resource SchemaURL: %s", rp.SchemaUrl())
+		buf.logAttributes("Resource attributes", rp.Resource().Attributes())
+		ilps := rp.ScopeProfiles()
+		for j := 0; j < ilps.Len(); j++ {
+			buf.logEntry("ScopeProfiles #%d", j)
+			ilp := ilps.At(j)
+			buf.logEntry("ScopeProfiles SchemaURL: %s", ilp.SchemaUrl())
+			buf.logInstrumentationScope(ilp.Scope())
+			profiles := ilp.Profiles()
+			for k := 0; k < profiles.Len(); k++ {
+				buf.logEntry("Profile #%d", k)
+				profile := profiles.At(k)
+				buf.logAttr("Profile ID", profile.ProfileID())
+				buf.logAttr("Start time", profile.StartTime().String())
+				buf.logAttr("End time", profile.EndTime().String())
+				buf.logAttributes("Attributes", profile.Attributes())
```

Comment:
Should we log the samples?

Reply:
@dmathieu that sounds good!
Should I loop through all samples from each profile though?
That would add an extra loop in a already 3 loops chain.

I was thinking about something like this:

```go
if ps := profile.Profile().Sample(); ps.Len() != 0 {
	for l := 0; l < ps.Len(); l++ {
		buf.logAttr("Sample at %d", l)
		buf.logAttr("Value", ps.At(l).Value())
	}
}
```

WDYT?

Reply:
I'm not sure how this is different from looping through each profile. Isn't that what it does?
Is it the amount of data possibly being logged that concerns you?

Reply:
Yep

Reply:
Do we usually get a lot of samples on each profile?

Reply:
The number of samples is `sampling frequency * number of cores * report interval`. So there can be many, yes.
But this is a debug exporter, and it will only be enabled if the required telemetry level is "detailed". So I'd be tempted to display their value anyway.

Reply:
I think for detailed level it makes sense to print the samples. In the normal level output (which would live here https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/debugexporter/internal/normal ) we can abbreviate it more

Reply:
@dmathieu would you have some time to take a look?
I've changed a couple of stuff to have all values printed (or most of it).